agent/containers/images/Dockerfile.layered.j2

```diff
@@ -7,13 +7,10 @@ ENTRYPOINT ["/container_entrypoint"]
 {% endif %}
 
 {% if kind in ('tools', 'all') %}
-{% if distro.startswith('fedora') or distro == 'centos-7' or distro == 'centos-8' %}
-COPY ./{{ distro }}-pcp.repo /etc/yum.repos.d/pcp.repo
-{% if distro.startswith('centos') %}
+{% if distro == 'centos-7' or distro == 'centos-8' %}
```

Comment:
Did you change the Prometheus conditionals deliberately? Isn't there a centos 9 that would match the old pattern?

Reply:
I think I did not change the _effect_ of the conditionals on Prometheus.  (I would welcome it if you could prove otherwise....)

I removed the `COPY` at (old) line 10 which left us with a nested conditional selecting any `centos` release, but, since the outer conditional selected only `centos-7` or `centos-8`, the inner one seemed redundant -- that is, the inner one only filtered out `fedora` releases, and so I removed that from the outer one instead.

So, I think the net result is no change...am I wrong?

Reply:
I was actually off my computer when you posted this and for the heck of it (correctly expecting a small review) I took the GitHub notification on my iPhone, where it's not as easy to examine the overall context. 😆 

So, yeah, you're right, it appears you haven't really changed the conditional. And we could I suppose question whether you *should* have, but making sure we support a "centos9" is likely to involve more than this case, so this probably isn't the right place or time. (Centos9 and RHEL 9 have been out for a while, and I was a bit surprised to discover we're apparently not supporting them in our Agent container builds ... but I believe that RHEL 9 was pre-release when Mustafa did the embedded PCP/Prometheus work.)

Reply:
> it appears you haven't really changed the conditional.

😅

We currently do build a containerized Agent for CentOS Stream 9 by default ([look here](https://quay.io/repository/pbench/pbench-agent-all-centos-9)).  The reason why this conditional doesn't reference `centos-9` is that that distro doesn't need special handling for Prometheus (see this [code comment](https://github.com/distributed-system-analysis/pbench/blob/01da8f29702700ed79f23edf5d9c554ef5be2164/agent/containers/images/Makefile#L240)).

RHEL containers are a different story.  While we apparently can freely distribute containers based on the RHEL UBI base image, it's less clear that we can distribute the dependencies which we install on top of it...so, until that's resolved, we don't build a containerized Agent using RHEL base images by default.  However, we do build RPMs for RHEL (at least in the CI), and most of the mechanism for building a containerized Agent using a UBI base image is in place (i.e., the build is all parameterized, but the UBI images don't come from Quay, so that needs to be dealt with), so it shouldn't (at least in theory...) be a huge effort to build a RHEL-based container.  However, anyone who wants to run the containerized Agent on RHEL has the option of running a CentOS-based container...I'm not sure what the advantage of running a RHEL-based container would be.